routes/user/index.js

```diff
@@ -0,0 +1,55 @@
+const express = require( 'express' )
+const router = express.Router()
+const passport = require( '../../configuration/passport' )
+const User = require( '../../database/user' )
+
+router.post( '/signup', ( request, response, next ) => {
+  const { email, password } = request.body
+
+  User.findByEmail( email )
+    .then( user => {
+      request.flash( 'message', 'Email already registered.' )
+      response.redirect( '/' )
+    })
+    .catch( error => {
```

Comment:
This is misleading. You're assuming that _all_ errors will be caused because the user couldn't be found.